### DISCUSSION TITLE
How to configure where eclipse.jdt.ls is located?

### DISCUSSION THREAD

**Discussion by carlbolduc** [joaotavora/eglot#788] [Q&A] [answered]
When I enable eglot in a Java project, I get the following message:
```
Path to eclipse.jdt.ls directory (could not find it in CLASSPATH)
```
I then provide the correct path:
```
~/.emacs.d/jdt-language-server-latest
```
Finally, I get the following message:
```
Add path to the server program to CLASSPATH environment variable?
```
I answer yes and eglot successfully connects to the jdt server.

However, then next time I open the same Java project, I get the same behaviour, I have to point eglot to the jdt.

How can I configure this in my init file so that I don't have to configure the location each time I open a Java project?

**Comment by mkcms** [comment]
> How can I configure this in my init file so that I don't have to configure the location each time I open a Java project?

I described a way to do it here:

https://github.com/joaotavora/eglot/issues/176#issuecomment-445021620

**Comment by mkcms** [comment]
> However, then next time I open the same Java project, I get the same behaviour, I have to point eglot to the jdt.

Do you mean after you restart Emacs? Because it should remember the path until you exit Emacs.

**Comment by carlbolduc** [comment]
> Do you mean after you restart Emacs? Because it should remember the path until you exit Emacs.

Yes, after I restart Emacs.

**Comment by easbarba** [comment, marked as answer]
You can go the traditional way, add it on your  `$CLASSPATH`:

`CLASSPATH="$CLASSPATH:$HOME/.emacs.d/eclipse.jdt.ls/plugins/org.eclipse.equinox.launcher_1.5.200.v20180922-1751.jar"`

or per project, of course.
